src/app/Shared/Services/Api.service.tsx

```diff
@@ -401,6 +401,12 @@ export class ApiService {
     return this.sendRequest(apiVersion, path, { method: 'GET' }).pipe(map(resp => resp.json()), concatMap(from), first());
   }
 
+  graphql<T>(query: string): Observable<T> {
+    const headers = new Headers();
+    headers.set('Content-Type', 'application/graphql');
+    return this.sendRequest('beta', 'graphql', { method: 'POST', body: query, headers }).pipe(map(resp => resp.json()), concatMap(from), first());
```

Comment:
Is it just me, or does smoketest with `BasicAuthManager` return a `401` for the graphql query?

Reply:
Actually, it looks like this same bug also affects the `Automated Rules` view, which makes a request in pretty much the same way. When I enable Basic auth I also get a 401 response when trying to create a rule. I'll file a separate bug.